src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -237,14 +227,30 @@ trait Implicits {
    */
   class ImplicitInfo(val name: Name, val pre: Type, val sym: Symbol) {
     private var tpeCache: Type = null
+    private var depolyCache: Type = null
```

Comment:
Isn't `private[this]` required to make sure that it uses the fields rather than accessors? (I'm thinking more about inline cache usage etc)

Reply:
Yes, `private[this]` access uses `GETFIELD` rather than `INVOKESPECIAL` here, but there is it no difference in performance on HotSpot as getter/setter methods are unconditionally inlined (even if `-XX:MaxInlineLevel` is reached.)